TreeView: add the "getting started" article (#2591)
DevExpress/devextreme-documentation: concepts/05 UI Components/TreeView/00 Getting Started with TreeView/30 Search Data.md

> The TreeView can display a search bar that allows users to search nodes. Enable the [searchEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) property to add the search bar. Use the [searchMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchMode) property to specify whether nodes should contain (default), start with, or match the search string. In this tutorial, search results include only nodes that start with the entered string:

Reviewer: > Use the [searchMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchMode) property to specify whether nodes should contain (default), start with, or match the **search string**. In this tutorial, search results include only nodes that start with the **entered string**:

I do not think it's a good idea to change the term introduced in the previous sentence.

```suggestion
The TreeView can display a search bar that allows users to search nodes. Enable the [searchEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) property to add the search bar. Use the [searchMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchMode) property to specify whether nodes should contain (default), start with, or match the search string. In this tutorial, search results include only nodes that start with the search string:
```

The TreeView can display a search bar that allows users to search nodes. Enable the [searchEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) property to add the search bar. Use the [searchMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchMode) property to specify whether nodes should contain (default), start with, or match the search string. In this tutorial, search results include only nodes that start with the entered string:

---
##### jQuery

    <!-- tab: index.js -->
    $(function(){
        $("#treeView").dxTreeView({ 
            // ...
            searchEnabled: true,
            searchMode: "startswith"
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-tree-view ...

>         searchEnabled="true"

Reviewer 2: ```suggestion
        [searchEnabled]="true"
```

        searchMode="startswith">
          <!-- ... -->
    </dx-tree-view>

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxTreeView ...
            :search-enabled="true"

>             search-mode="startswith"
>         />

Reviewer 2: ```suggestion
            search-mode="startswith">
            <!-- ... -->
        </DxTreeView>
```

>             search-mode="startswith"
>         >

Reviewer 2: ```suggestion
            search-mode="startswith">
```

> Enable the [searchEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) property to add the search bar. Use the [searchMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchMode) property to specify whether items should contain (default), start with, or match the search string. In this tutorial, the search mode is *"startswith"*:

Reviewer 2: Rename the article to Search Nodes.

> Enable the [searchEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) property to add the search bar.

You get to "enabling" too fast. Describe the functionality first, slower the pace.

```suggestion
The TreeView can display a search bar that allows users to search nodes. Enable the [searchEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) property to add the search bar. Use the [searchMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchMode) property to specify whether nodes should contain (default), start with, or match the search string. In this tutorial, search results include only nodes that start with the entered string:
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function(){
        $("#treeView").dxTreeView({ 
            // ...
            searchEnabled: true,
            searchMode: "startswith"
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-tree-view ...

>         searchEnabled=true

Reviewer 2: ```suggestion
        searchEnabled="true"
```

        searchMode="startswith"

>     >

Reviewer 2: ```suggestion
        searchMode="startswith">
```

          <!-- ... -->
    </dx-tree-view>

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxTreeView ...
            :search-enabled="true"
            search-mode="startswith"

>         />

Reviewer 2: Don't use self-closing tags if there were (or will be) somehing inside them. In this tutorial, it was a template.

    </template>
    // ...

##### React

    <!-- tab: App.js -->
    import React from 'react';

    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import TreeView from 'devextreme-react/tree-view';
    import products from './products';

> 

Reviewer 2: ```suggestion
    // ...
```

    function App() {
        // ...

> 

Reviewer 2: ```suggestion
```
